query/lexer.go

```diff
@@ -58,6 +69,20 @@ func tokenize(query string) ([]token, error) {
 			continue
 		}
 
+		if len(split) >= 2 && split[0] == "metadata" {
```

Comment:
The lexer should not be aware of a particular keyword meaning. Your extension to the query language means that the lexer should now be able to read tokens with one key and two values (that you called qualifier and subqualifier: I guess that make sense, I wonder if there is already an established term for that?). What those token means is the responsibility of the parser (the next step).

You likely noticed `newTokenSearch` which already has some semantic notion. This is not great, but let's try to not follow on this path. Ideally the lexer should only care about breaking a string into tokens.

Could you adjust your code with this optic?

Reply:
Maybe we need a `newTokenKVV` or something like that?

Reply:
Yes, that makes sense. I did both.